src/ShellCheck/Checks/Commands.hs

```diff
@@ -607,9 +607,11 @@ prop_checkInteractiveSu1 = verify checkInteractiveSu "su; rm file; su $USER"
 prop_checkInteractiveSu2 = verify checkInteractiveSu "su foo; something; exit"
 prop_checkInteractiveSu3 = verifyNot checkInteractiveSu "echo rm | su foo"
 prop_checkInteractiveSu4 = verifyNot checkInteractiveSu "su root < script"
+prop_checkInteractiveSu5 = verify checkInteractiveSu "su -c something"
```

Comment:
This unit test makes sure you *do* get a warning :P 

You can use `verifyNot` to make sure you don't